Minor refactoring

todorpashkulov opened the pull request.

todorpashkulov committed:
Change Log:
-refactored Generala(Removed singleton,moved load properties )
-added Printer and Constants classes

todorpashkulov committed:
Change Log:
-renamed utils package to generalahelpers
-updated GeneralaUtils names and moved to generalahelpers package.

todorpashkulov committed:
Change Log:
-renamed Combinations methods

todorpashkulov committed:
Change Log:

todorpashkulov committed:
Change Log:
-refactored GeneralConstants,Combinations,GeneralaPrinter
-updated Combinations.addStraight()

todorpashkulov committed:
Change Log:
-moved findCombinations from Generala to CombinationFinder
-moved Main to application package
-renamed generalautils package into utils

todorpashkulov committed:
Refactored most of the classes,renamed local variables,changed method signature

todorpashkulov committed:
Change Log:
-refactored CombinationFinder and GeneralaPrinter to utils
-refactored hasStraight() method in GeneralaFinder
-refactored names in Combination finder

LazarN commented on src/generala/constants/GeneralaConstants.java:
```diff
@@ -0,0 +1,12 @@
+
+public class GeneralaConstants {
+
+    public static final int PAIR_SIZE = 2;
```
Remove _SIZE from constants name.

LazarN commented on src/generala/utils/CombinationFinder.java:
```diff
@@ -0,0 +1,240 @@
+        resetInstanceVariablesForCombinationCreating();
+    }
+
+    private void resetInstanceVariablesForCombinationCreating() {
```
Perhaps there is a better name for this method. `resetCombinationVariables` you do not need to put anything java specific in method names. The person who reads the code is aware that these are instance variables.

LazarN replied:
Avoid using Java related terms in method signature. Method signature should be clear enough to be read by everyone. Imagine that non dev person will read this. I doubt everyone knows what is instance variable.
Better name would be `resetCombinationVariables()`

LazarN commented on src/generala/utils/CombinationFinder.java:
```diff
@@ -0,0 +1,240 @@
+        List<Integer> rolledSidesList = new ArrayList<>(rolledSidesSet);
+        for (int i = 0, counter = 0; i < rolledSidesList.size() - 1; i++) {
+            if (rolledSidesList.get(i) - 1 != rolledSidesList.get(i + 1)) {
+                counter = 0;
```
Why do you need the counter? Obviously the method does more than one thing - finding if the combination is straight and incrementing counter. Try to separate it if you rely on the counter variable (apply this only if it is possible)

Can we use it without the counter:



```
public boolean hasStraight(Set<Integer> rolledSidesSet) {
        List<Integer> rolledNumbers = new ArrayList<>(rolledSidesSet);
        Collections.sort(rolledNumbers);
        for (int i = 0; i < rolledNumbers.size() - 1; i++) {
            if (rolledNumbers.get(i) + 1 != rolledNumbers.get(i + 1)) {
                return false;
            }
        }

        return true;
    }
```

LazarN commented on src/generala/utils/CombinationFinder.java:
```diff
@@ -0,0 +1,240 @@
+        return rolledCombinations.contains(combinationEnum);
+    }
+
+    private boolean hasStraight(Set<Integer> rolledSidesSet) {
```
Can we use it without the counter variable?

 ```
public boolean hasStraight(Set<Integer> rolledSidesSet) {
       List<Integer> rolledNumbers = new ArrayList<>(rolledSidesSet);
        Collections.sort(rolledNumbers);

        for (int i = 0; i < rolledNumbers.size() - 1; i++) {
            if (rolledNumbers.get(i) + 1 != rolledNumbers.get(i + 1)) {
                return false;
            }
        }

        return true;
    }
```

LazarN commented on src/generala/utils/CombinationFinder.java:
```diff
@@ -0,0 +1,240 @@
+        return false;
+    }
+
+    private boolean canBreakEarly(int playerRolledCombinationCount, int combinationTreeMapSize) {
```
Avoid using Java key words in method signature and variable names. In this case there must be a better name for this method. Applicable for the method name from previous comment. Imagine that non dev person reads your method signature and has to understand what action is described only by reading it.

LazarN commented on src/generala/utils/GeneralaUtils.java:
```diff
@@ -0,0 +1,65 @@
+        }
+    }
+
+    private static void addCurrentSideToDiceRollMapWithDuplicates(DiceRoll diceRoll, int value) {
```
Method name is really long. If I encounter this method name in some code snippet  I will not even read the body.
`addCurrentSideToDiceRoll()` is descriptive enough.

LazarN commented on out/production/GeneralaGame/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
```
Read about .gitignore file. We definitely need one here. These kind of files are not supposed to be part of the source.
Jar, .MF, .class, I will add one for this repo in master branch. Keep in mind that this file is used for ignoring all files that are not needed.

todorpashkulov committed:
Change Log:
-fixed most of the review problems
-left two reviews unsolved for further discussion

LazarN commented:
Mege the pull request into master.
